Added google_apigee_api_deployment resource

apichick opened the pull request:
This is a PR to add the google_apigee_api_deployment resource

```release-note:new-resource
`google_apigee_api_deployment`
```

roaks3 commented on mmv1/third_party/terraform/website/docs/r/apigee_api_deployment.html.markdown:
```diff
@@ -0,0 +1,108 @@
+An API proxy deployment can be imported using any of these accepted formats:
+
+* `organizations/{{org_id}}/environments/{{environment}}/apis/{{api_id}}/revisions/{{revision}}/deployments/{{name}}`
+* `{{org_id}}/{{environment}}/{{sharedflow_id}}/{{revision}}/{{name}}`
```
s/sharedflow_id/api_id/ ?

Similar for various shared flow references below this on the page.

apichick replied:
updated the references

roaks3 reviewed (commented):
Overall LGTM, but a couple of high-level questions:
- Does this actually need to be a handwritten resource, or could it be implemented as an mmv1 resource like `mmv1/products/apigee/Environment.yaml`? I didn't notice anything that needed to be handwritten here, and it looked like the sharedflow resource needed `sendRequestRawBodyWithTimeout`, which I don't see used here.
- Do you know why the test needs to be skipped in VCR?

apichick commented:
Question, I did it not use it using yamls, because the google_apigee_sharedflow_deployment resource which is strcturally similar was not done like that, and I assumed that was the case, because we need to use the same text-fixtures as with google_apigee_api resource. Is it possible to implement resources with yaml and use test-fixtures, where would you place them?

apichick commented:
I modified the PR so now there resource is done using YAML. However i introduced a change in the google_apigee_api resource and I did that for a reason. Originally I was passing to the google_apigee_api resource the org_id as "{{org_name}}" , for consistency, just as the google_apigee_shareflow resource did. However I realized that forces to declare the dependency of the organization explicitly. So I decided to pass org_id as "organizations/{{org_name}}". I think that should modified in the google_apigee_sharedflow resource too. If that change is not OK because it is backwards incompatible I can drop it, however I think it is better this way.

roaks3 reviewed (commented):
Just posting a comment to clear the reminders: this is waiting on the feasibility of supporting both "{{org_name}}" and "organizations/{{org_name}}" formats. Removing the existing format would be considered a breaking change.

osakri commented:
@roaks3 any updates on this ?
this resource could be very helpful for our infra.

cjhouser commented:
@roaks3 what did the feasibility study determine?

roaks3 commented:
This PR was closed due to inactivity. Please add future comments regarding this feature on https://github.com/hashicorp/terraform-provider-google/issues/22581 (I'm currently not aware of any timeline for this feature).
